src/core/properties/cameraproperty.cpp

```diff
@@ -244,20 +217,55 @@ CameraProperty::operator const Camera&() const { return *camera_; }
 
 CameraProperty* CameraProperty::clone() const { return new CameraProperty(*this); }
 
-void CameraProperty::resetCamera() {
-    NetworkLock lock(this);
+bool CameraProperty::changeCamera(const std::string& name) {
+    if (name != camera_->getClassIdentifier()) {
+        NetworkLock lock(this);
+        auto newCamera = factory_->create(name);
+        hideConfiguredProperties();
+
+        newCamera->updateFrom(camera_.get());
+        camera_ = std::move(newCamera);
+        camera_->configureProperties(this);
 
-    lookFrom_.resetToDefaultState();
-    lookTo_.resetToDefaultState();
-    lookUp_.resetToDefaultState();
+        modifedConfiguredProperties();
 
-    // Update template value
-    camera_->setLookFrom(lookFrom_.get());
-    camera_->setLookTo(lookTo_.get());
-    camera_->setLookUp(lookUp_.get());
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void CameraProperty::hideConfiguredProperties() {
+    for (auto& p : ownedCameraProperties_) p->setVisible(false);
+}
+
+void CameraProperty::modifedConfiguredProperties() {
+    for (auto p : cameraProperties_) {
+        p->propertyModified();
+    }
+}
+
+CameraProperty& CameraProperty::setCamera(const std::string& cameraIdentifier) {
+    cameraType_.setSelectedIdentifier(cameraIdentifier);
+    return *this;
+}
+
+CameraProperty& CameraProperty::setCamera(std::unique_ptr<Camera> newCamera) {
+    if (newCamera) {
+        NetworkLock lock(this);
+        hideConfiguredProperties();
```

Comment:
Should the previous visibility state be restored in the end here?

Reply:
Configure will show them again